### PR TITLE
Add "make your selections" alert  to Notification Settings

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
This PR adds an alert at the top of the Notification Settings page when there are notification options that the user has not yet opted into or out of. The alert contains a jump link that scrolls to the first radio button group that needs the user's attention. The jump link target will update as the user makes selections and the alert will be removed from the screen after the user no longer has notification options that they need to opt into or out of.

In order to make this work, I needed to create a new selector function, `selectChannelsWithoutSelection()`, that would easily surface the channels that are available to the user (filtering out channels the user is not eligible to see because they are not a VA patient or lack the required contact info) which they have not previously made a selection for. This in turn required making a number of other supporting selectors.

## Original issue(s)
department-of-veterans-affairs/va.gov-team#0000

## Testing done
- started by building the new cypress tests first
- built the new redux selectors with TDD

## Screenshots
jump-link

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx [hunk 1]
```diff
@@ -99,15 +146,11 @@ const NotificationSettings = ({
             // we handle the health care group a little differently
             // TODO: I don't like this check. what does `group3` even mean?
             if (groupId === 'group3') {
-              if (!isPatient) {
-                return null;
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx:103]: logic for showing/hiding the health care group has been moved to the redux selectors. and we have existing Cypress tests to ensure that the group is removed for non-patients.

File: src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx [hunk 2]
```diff
@@ -27,17 +31,48 @@ import HealthCareGroupSupportingText from './HealthCareGroupSupportingText';
...
+      >
+        <div>
+          <p>
+            We’ve added notification options to your profile. Tell us how you’d
+            like us to contact you.
+          </p>
+          <p>
+            <a
+              href={`#${firstChannelId}`}
+              // className="vads-u-text-decoration--none vads-u-padding--1 vads-u-margin-left--neg1 vads-u-margin-top--neg1 vads-u-margin-bottom--neg1"
```
Comment (micahchiang) [inline src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx:50]: Can these classes just be removed or are they needed later?

**Review by tmitchellgcio — APPROVED** [review]

**Review by micahchiang — COMMENTED** [review]
I couldn't tell from the attached gif, does focus change to the jump link target when a user clicks on the banner?

File: src/applications/personalization/profile/components/notification-settings/NotificationRadioButtons.jsx [hunk 3]
```diff
@@ -206,7 +206,7 @@ const NotificationRadioButtons = ({
   );
 
   return (
-    <fieldset className={fieldsetClass} disabled={disabled}>
+    <fieldset className={fieldsetClass} disabled={disabled} id={id}>
```
Comment (timwright12) [inline src/applications/personalization/profile/components/notification-settings/NotificationRadioButtons.jsx:209]: TIL you can disable a fieldset

**Review by timwright12 — APPROVED** [review]

File: src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx (hunk 2 above)
Comment (erikphansen) [reply src/applications/personalization/profile/components/notification-settings/NotificationSettings.jsx:50]: oops, thanks for catching that. I actually do want those and had just commented them out for a quick test.

**Pull request merged**